src/sharepoint/rest/rest.ts

```diff
@@ -10,6 +11,16 @@ import { Queryable } from "./queryable";
  */
 export class Rest {
 
+
+    /**
+     * Executes a search against this web context
+     * 
+     * @param query The SearchQuery definition
+     */
+    public search(query: SearchQuery): SearchResult {
+        return new Search("_api/search", query);
```

Comment:
Can you update this  to execute the query and return a promise? So signature would be the same except:
`public search(query: SearchQuery): Promise<SearchResult>`
See note on "get" related to this.

That would allow folks to just call pnp.sp.search({stuff}).then(...).